### DISCUSSION TITLE
Automatically output build failure in detail in ci

### DISCUSSION THREAD

**Discussion by msclock** [microsoft/vcpkg#42509] [Q&A] [answered]
Hi, community. 

Normally, the vcpkg build is passing as expected. But some cases which has no local building environment brings unexpected failure in ci. For example, https://github.com/msclock/cppcheck-wheel/actions/runs/12085636523/job/33703271748#step:6:1018

It seems to output failure into the file `${CMAKE_BINARY_DIR}/vcpkg_installed/vcpkg/issue_body.md` in detail 

Is there one way to output the failure in detail in ci automaically.

**Comment by Osyotr** [comment]


**Reply by dg0yt** [reply]
I think the question is about using vcpkg in CI, not about vcpkg's CI.

**Comment by msclock** [comment]
Seems to need some patch changes into `VCPKG_ROOT/scripts/buildsystems/vcpkg.cmake` before include the toolchain file.

**Reply by msclock** [reply]
Now, my workaround is to add a patch function to message faultre in detail. See https://github.com/serious-scaffold/ss-cmake/blob/ab17e79610f249422663c0f3c70d5d4a70a0ea15/template/cmake/vcpkg/bootstrap/vcpkg_configure.cmake#L55-L77.

**Reply by msclock** [reply]
As the below discussion, Check the existance of `issue_body.md` and `cat` it in CI is another way.

**Comment by dg0yt** [comment, marked as answer]
You might simply change your CI script/action to process `${CMAKE_BINARY_DIR}/vcpkg_installed/vcpkg/issue_body.md` when a build step failed.

**Reply by msclock** [reply]
This workaround recognizes vcpkg install failure inaccurately in a build step. ):

**Reply by dg0yt** [reply]
> This workaround recognizes vcpkg install failure inaccurately in a build step. ):

Unclear.

**Reply by msclock** [reply]
> > This workaround recognizes vcpkg install failure inaccurately in a build step. ):
> 
> Unclear.

Sorry for my unclear expression. 

> recognizes vcpkg install failure inaccurately

I consider the cmake configuration may fail in another ways instead of `vcpkg install` failure.

**Reply by dg0yt** [reply]
> I consider the cmake configuration may fail in another ways instead of `vcpkg install` failure.

But then, the file will not exist (at least in a clean CI run).

**Reply by msclock** [reply]
> the file will not exist

Yep, this is the simplest way. Check the existance of `issue_body.md` and `cat` it in CI.
